lib/circuit.js

```diff
@@ -222,6 +232,17 @@ class CircuitBreaker extends EventEmitter {
         handleError(error, this, timeout, args, resolve, reject);
       }
     });
+    if (this.options.cache) {
```

Comment:
Is it overkill to allow the option to be either a `boolean`, an `Object`, a `Map` or a `WeakMap`? This would allow users to create a circuit with a pre-primed cache. 

As I'm writing this, I am leaning towards that being a feature, but perhaps a feature for later... :)